[WIP] US - Credential Validator

loqo71la opened the pull request.

Commit:
update dependencies gradle

Commit:
Add negative case for validate credentials.

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+    private static final String NULL_MESSAGE = "Can not validate a null credential"
+
+    /**
+     * Validate a encrypt credential on Enforce.
```
Validates ...

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+     * @return true if it's possible validate the credential, otherwise return false.
+     * @exception IllegalArgumentException if credential is null
+     */
+    public static boolean validate(Credential credential) {
```
isValidCredential(...)

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    /**
+     * Return the decrypted credential.
```
Returns

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+            throw new IllegalArgumentException(NULL_MESSAGE)
+        }
+        credential = decryptCredential(credential)
+        Connector connector = new Connector(credential.loginFormat)
```
connector variable is not necessary

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+        credential = decryptCredential(credential)
+        Connector connector = new Connector(credential.loginFormat)
+        try {
+            connector.login(credential)
```
this statement would be enough 

(new Connector(credential.loginFormat)).login(credential)

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+        Connector connector = new Connector(credential.loginFormat)
+        try {
+            connector.login(credential)
+        } catch (Exception ex) {
```
the Login implementation just throw general exceptions?

loqo71la replied:
Yes, Connector.groovy line 60

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialValidator.groovy:
```diff
@@ -0,0 +1,53 @@
+import java.nio.file.Paths
+
+/**
+ * Validate credentials registered on Enforce.
```
This class performs/validates SFDC credentials

JimmyRojas commented:
reviewed.. :)
@marcocdlv, please take a look too

Commit:
Rename validate method by isValidCredential and remove unnecessary variables

Commit:
Add new credentials and extract unnecessary logic from CredentialValidatorTest

Commit:
Implement logic for validate encrypt/decrypt credentials in the followings classes:
- CredentialsValidatorTest.groovy: Add test for validate encrypt/decrypt credentials
- CredentialsValidator.groovy: Add validateCredentials when using connector
- credentials.dat: Add new credentials
- Credential.groovy: Add equals and hashcode methods

loqo71la commented:
@JimmyRojas @marcocdlv please review my changes

Commit:
show credentials with validation

RockFercho commented:
@JimmyRojas @marcocdlv  please check my changes

Commit:
Remove unnecessary imports and sentences

Commit:
Merge branch 'US_CREDENTIAL_VALIDATOR' of github.com:RockFercho/enforce-gradle-plugin into US_CREDENTIAL_VALIDATOR

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialGiver.groovy:
```diff
@@ -42,16 +50,40 @@ class CredentialGiver extends CredentialManagerTask {
+                printCredential(credential)
+            }
+            else {
+
```
remove this empty line

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialGiver.groovy:
```diff
@@ -42,16 +50,40 @@ class CredentialGiver extends CredentialManagerTask {
+                        printCredential(credential)
+                        logger.quiet("Status: $VALID_STATUS_MESSAGE")
+                    }
+                }
```
} catch (Exception e) {

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/wsc/Credential.groovy:
```diff
@@ -19,4 +19,31 @@ class Credential {
+
+        Credential that = (Credential) o
+
+        if (id != that.id) return false
```
us OR operator instead of many IF statements

JimmyRojas commented on src/main/groovy/org/fundacionjala/gradle/plugins/enforce/tasks/credentialmanager/CredentialManagerTask.groovy:
```diff
@@ -14,9 +16,14 @@ import java.nio.file.Paths
 
     public CredentialManagerInput credentialManagerInput
     public String location = 'home'
+    public String status = ""
```
make it protected and expose it through getter method

JimmyRojas commented:
see some added comments

JimmyRojas commented:
I'm still waiting for the scenarios that we've agreed for UT
